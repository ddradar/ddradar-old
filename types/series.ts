export type Series =
  | '1st'
  | '2ndMIX'
  | '3rdMIX'
  | '4thMIX'
  | '5thMIX'
  | 'DDRMAX'
  | 'DDRMAX2'
  | 'EXTREME'
  | 'SuperNOVA'
  | 'SuperNOVA2'
  | 'X'
  | 'X2'
  | 'X3 VS 2ndMIX'
  | '2013'
  | '2014'
  | 'A'
  | 'A20'

export const SeriesList: Series[] = [
  '1st',
  '2ndMIX',
  '3rdMIX',
  '4thMIX',
  '5thMIX',
  'DDRMAX',
  'DDRMAX2',
  'EXTREME',
  'SuperNOVA',
  'SuperNOVA2',
  'X',
  'X2',
  'X3 VS 2ndMIX',
  '2013',
  '2014',
  'A',
  'A20'
]

export function getSeriesName(series: Series) {
  return series === 'DDRMAX' || series === 'DDRMAX2' ? series : `DDR ${series}`
}

export function isSeries(object: unknown) {
  return (
    typeof object === 'string' && (SeriesList as Array<string>).includes(object)
  )
}
