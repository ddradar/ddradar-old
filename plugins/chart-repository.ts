import { Level } from '~/types/level'
import { PlayStyle } from '~/types/play-style'
import { isStepChart, StepChart } from '~/types/step-chart'

export const chartVersion = 20200127

export const fetchSongCharts = async (
  songId: string,
  useMaster: boolean = false
) =>
  (await fetchChartJson(useMaster))
    .filter((c) => c.songId === songId)
    .sort((l, r) =>
      l.playStyle !== r.playStyle
        ? l.playStyle - r.playStyle
        : l.difficulty - r.difficulty
    )

export const fetchChartsByLevel = async (
  playStyle: PlayStyle,
  level: Level,
  useMaster: boolean = false
) =>
  (await fetchChartJson(useMaster))
    .filter((c) => c.level === level && c.playStyle === playStyle)
    .sort((l, r) =>
      l.songName < r.songName
        ? -1
        : l.songName > r.songName
        ? 1
        : l.difficulty - r.difficulty
    )

const masterUrl = 'https://staging.ddradar.app'
const fetchChartJson = async (useMaster: boolean) => {
  const jsonUrl = `${useMaster ? masterUrl : ''}/chart.json`
  const jsonData = await (await fetch(jsonUrl)).json()
  if (Array.isArray(jsonData) && isStepChart(jsonData[0]))
    return jsonData as StepChart[]
  throw new Error('JSON file is not StepChart[]')
}
