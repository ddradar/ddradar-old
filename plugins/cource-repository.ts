import courceJson from '~/static/cource.json'
import { Cource, isCource } from '~/types/cource'
import { Difficulty, getDifficultyName } from '~/types/difficulty'
import { getPlayStyleName, PlayStyle } from '~/types/play-style'

export const version = 20190612

export const fetchCource = (
  courceId: string,
  playStyle: PlayStyle,
  difficulty: Difficulty
): Cource => {
  const cource = courceJson.find((c) => c.id === courceId)
  if (cource === undefined) throw new Error(`Not Found Cource Id: ${courceId}`)
  const order = cource.orders.find(
    (o) => o.playStyle === playStyle && o.difficulty === difficulty
  )
  if (order === undefined)
    throw new Error(
      `Not Found ${getPlayStyleName(playStyle)}/${getDifficultyName(
        difficulty
      )} order`
    )
  return {
    id: cource.id,
    name: cource.name,
    playStyle: order.playStyle,
    difficulty: order.difficulty,
    level: order.level,
    chartOrder: order.chartOrder
  } as Cource
}

export const fetchAllCourceList = () =>
  courceJson
    .flatMap((c) =>
      c.orders.map((o) => ({
        ...o,
        id: c.id,
        name: c.name
      }))
    )
    .filter((c) => isCource(c))
