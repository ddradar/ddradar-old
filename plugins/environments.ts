import { Plugin } from '@nuxt/types'
import { config } from 'dotenv'

config()

export type EnvironmentVariables = {
  NODE_ENV: string
  FIREBASE_API_KEY: string
  FIREBASE_AUTH_DOMAIN: string
  FIREBASE_DATABASE_URL: string
  FIREBASE_PROJECT_ID: string
  FIREBASE_STORAGE_BUCKET: string
  FIREBASE_MESSAGING_SENDER_ID: string
  FIREBASE_APP_ID: string
}

/* eslint-disable no-process-env */
export const environments: EnvironmentVariables = {
  NODE_ENV: process.env.NODE_ENV!,
  FIREBASE_API_KEY: process.env.FIREBASE_API_KEY!,
  FIREBASE_AUTH_DOMAIN: process.env.FIREBASE_AUTH_DOMAIN!,
  FIREBASE_DATABASE_URL: process.env.FIREBASE_DATABASE_URL!,
  FIREBASE_PROJECT_ID: process.env.FIREBASE_PROJECT_ID!,
  FIREBASE_STORAGE_BUCKET: process.env.FIREBASE_STORAGE_BUCKET!,
  FIREBASE_MESSAGING_SENDER_ID: process.env.FIREBASE_MESSAGING_SENDER_ID!,
  FIREBASE_APP_ID: process.env.FIREBASE_APP_ID!
}
/* eslint-enable no-process-env */

/** Validate environments values. */
export const validateEnvironments = ():
  | { valid: true }
  | { valid: false; keys: Extract<keyof EnvironmentVariables, string>[] } => {
  const invalidKeys: string[] = Object.keys(environments).filter((key) => {
    const value: unknown = (environments as any)[key]
    return value === undefined || value === null
  })
  return invalidKeys.length === 0
    ? { valid: true }
    : {
        valid: false,
        keys: invalidKeys as Extract<keyof EnvironmentVariables, string>[]
      }
}

declare module 'vue/types/vue' {
  interface Vue {
    $environments: EnvironmentVariables
  }
}

const environmentsPlugin: Plugin = (_, inject) => {
  inject('environments', environments)
}

export default environmentsPlugin
