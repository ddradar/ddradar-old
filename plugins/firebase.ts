import { Context, Plugin } from '@nuxt/types'
import firebase from 'firebase/app'

const firebasePlugin: Plugin = ({ env }: Pick<Context, 'env'>, _) => {
  if (!firebase.apps.length) {
    firebase.initializeApp({
      apiKey: env.apiKey,
      authDomain: env.authDomain,
      databaseURL: env.databaseURL,
      projectId: env.projectId,
      storageBucket: env.storageBucket,
      messagingSenderId: env.messagingSenderId,
      appId: env.appId
    })
  }
}

export default firebasePlugin
