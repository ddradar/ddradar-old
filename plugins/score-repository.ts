import 'firebase/firestore'

import firebase from '~/plugins/firebase'
import { isDifficulty } from '~/types/difficulty'
import { isPlayStyle } from '~/types/play-style'
import { StepChart } from '~/types/step-chart'
import {
  hasNumberProperty,
  hasProperty,
  hasStringProperty
} from '~/utils/type-assert'

const db = firebase.firestore()

type ClearLamp =
  | 'NoPlay'
  | 'Failed'
  | 'Assisted'
  | 'Clear'
  | 'Life4'
  | 'GoodFC'
  | 'FC'
  | 'PFC'
  | 'MFC'

export const clearLampList: ClearLamp[] = [
  'NoPlay',
  'Failed',
  'Assisted',
  'Clear',
  'Life4',
  'GoodFC',
  'FC',
  'PFC',
  'MFC'
]

type DanceLevel =
  | '-'
  | 'E'
  | 'D'
  | 'D+'
  | 'C-'
  | 'C'
  | 'C+'
  | 'B-'
  | 'B'
  | 'B+'
  | 'A-'
  | 'A'
  | 'A+'
  | 'AA-'
  | 'AA'
  | 'AA+'
  | 'AAA'

export const rankList: DanceLevel[] = [
  '-',
  'E',
  'D',
  'D+',
  'C-',
  'C',
  'C+',
  'B-',
  'B',
  'B+',
  'A-',
  'A',
  'A+',
  'AA-',
  'AA',
  'AA+',
  'AAA'
]

export type Score = Pick<StepChart, 'songId' | 'playStyle' | 'difficulty'> & {
  score: number
  exScore: number
  clearlamp: ClearLamp
  rank: DanceLevel
  isPublic: boolean
}

export const getUserScore = async (uid: string, songId: string) => {
  const scores: Score[] = []
  const data = await db
    .collection(`version/1/users/${uid}/scores/`)
    .where('songId', '==', songId)
    .orderBy('playStyle')
    .orderBy('difficulty')
    .get()
  data.forEach((doc) => {
    const score = doc.data()
    if (isScore(score)) scores.push(score)
  })
  return scores
}

export const setUserScore = async (
  uid: string,
  isPublic: boolean,
  score: Score
) => {
  const doc = db.doc(
    `version/1/users/${uid}/scores/${score.songId}-${score.playStyle}-${score.difficulty}`
  )
  await db.runTransaction(async (tran) => {
    const prevDoc = await tran.get(doc)
    const prevScore = prevDoc.data()
    if (!isScore(prevScore)) {
      tran.set(doc, { ...score, createdAt: firebase.firestore.Timestamp.now() })
      return
    }
    const newScore = {
      ...prevScore,
      score: prevScore.score > score.score ? prevScore.score : score.score,
      exScore:
        prevScore.exScore > score.exScore ? prevScore.exScore : score.exScore,
      clearLamp: compareClearLamp(prevScore.clearlamp, score.clearlamp),
      rank: score.rank !== 'E' ? getDanceLevel(score.score) : score.rank,
      isPublic,
      updateAt: firebase.firestore.Timestamp.now()
    }
    tran.update(doc, newScore)
  })
}

const compareClearLamp = (left: ClearLamp, right: ClearLamp): ClearLamp =>
  clearLampList.indexOf(left) > clearLampList.indexOf(right) ? left : right

export const isScore = (obj: unknown): obj is Score =>
  typeof obj === 'object' &&
  obj !== null &&
  hasProperty(obj, 'playStyle', 'difficulty', 'isPublic') &&
  isPlayStyle(obj.playStyle) &&
  isDifficulty(obj.difficulty) &&
  typeof obj.isPublic === 'boolean' &&
  hasNumberProperty(obj, 'score', 'exScore') &&
  hasStringProperty(obj, 'songId', 'clearLamp', 'rank') &&
  (clearLampList as string[]).includes(obj.clearLamp) &&
  (rankList as string[]).includes(obj.rank)

export const getDanceLevel = (score: number): DanceLevel => {
  if (!Number.isInteger(score) || score < 0 || score > 1000000)
    throw new Error('score is invalid')
  return score < 550000
    ? 'D'
    : score < 590000
    ? 'D+'
    : score < 600000
    ? 'C-'
    : score < 650000
    ? 'C'
    : score < 690000
    ? 'C+'
    : score < 700000
    ? 'B-'
    : score < 750000
    ? 'B'
    : score < 790000
    ? 'B+'
    : score < 800000
    ? 'A-'
    : score < 850000
    ? 'A'
    : score < 890000
    ? 'A+'
    : score < 900000
    ? 'AA-'
    : score < 950000
    ? 'AA'
    : score < 990000
    ? 'AA+'
    : 'AAA'
}

const isPositiveInteger = (num: number) => Number.isInteger(num) && num >= 0

export const calcScore = (
  {
    notes,
    freezeArrow,
    shockArrow
  }: Pick<StepChart, 'notes' | 'freezeArrow' | 'shockArrow'>,
  marvelous: number,
  perfect: number,
  great: number,
  good: number,
  ok: number
) => {
  if (!isPositiveInteger(marvelous))
    throw new Error(`invalid marvelous: ${marvelous}`)
  if (!isPositiveInteger(perfect))
    throw new Error(`invalid perfect: ${perfect}`)
  if (!isPositiveInteger(great)) throw new Error(`invalid great: ${great}`)
  if (!isPositiveInteger(good)) throw new Error(`invalid good: ${good}`)
  if (!isPositiveInteger(ok)) throw new Error(`invalid ok: ${ok}`)

  const noteObjects = notes + freezeArrow + shockArrow
  const baseScore = 1000000 / noteObjects
  const rawScore =
    baseScore * (marvelous + ok) +
    (baseScore - 10) * perfect +
    (baseScore * 0.6 - 10) * great +
    (baseScore * 0.2 - 10) * good
  return Math.floor(rawScore / 10) * 10
}

export const calcExScore = (
  marvelous: number,
  perfect: number,
  great: number,
  ok: number
) => {
  if (!isPositiveInteger(marvelous))
    throw new Error(`invalid marvelous: ${marvelous}`)
  if (!isPositiveInteger(perfect))
    throw new Error(`invalid perfect: ${perfect}`)
  if (!isPositiveInteger(great)) throw new Error(`invalid great: ${great}`)
  if (!isPositiveInteger(ok)) throw new Error(`invalid ok: ${ok}`)

  return marvelous * 3 + ok * 3 + perfect * 2 + great * 1
}
