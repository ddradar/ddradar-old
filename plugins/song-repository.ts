import firebase from '@/plugins/firebase'
import 'firebase/firestore'
import { isSong, Song } from '@/types/song'

const db = firebase.firestore()

export async function fetchSongs(fieldName: string, condition: any) {
  const query = db
    .collection('version/1/songs')
    .where(fieldName, '==', condition)

  const querySorted =
    fieldName === 'nameIndex'
      ? query.orderBy('nameKana')
      : query.orderBy('nameIndex').orderBy('nameKana')
  const snapShot = await querySorted.get()
  return snapShot.docs.map(doc => doc.data()).filter(d => isSong(d)) as Song[]
}

export async function fetchSongById(songId: string) {
  const doc = await db.doc(`version/1/songs/${songId}`).get()
  const unknown = doc.data()
  return isSong(unknown)
    ? unknown
    : Promise.reject(
        new Error(`"version/1/songs/${songId}" is not Song object`)
      )
}
