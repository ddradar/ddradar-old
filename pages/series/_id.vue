<template>
  <section class="section">
    <h1 class="title">{{ pageTitle }}</h1>
    <div class="buttons">
      <b-button
        v-for="(series, index) in seriesList"
        :key="index"
        :to="`/series/${index}`"
        type="is-info"
        tag="nuxt-link"
        :disabled="index == selected"
        :outlined="index == selected"
      >
        {{ series }}
      </b-button>
    </div>
    <p class="subtitle">{{ message }}</p>
    <song-list v-if="selected != null" :loading="isLoading" :songs="songs" />
  </section>
</template>

<script lang="ts">
import { Context } from '@nuxt/types'
import { Vue, Component } from 'nuxt-property-decorator'
import { Song } from '@/types/song'
import { SeriesList, getSeriesName } from '@/types/series'
import { fetchSongs } from '@/plugins/song-repository'

@Component({
  components: {
    SongList: () => import('~/components/SongList.vue')
  }
})
export default class SeriesPage extends Vue {
  selected: number | null = null
  songs: Song[] = []
  isLoading = true
  seriesList = SeriesList

  public get pageTitle() {
    if (
      this.selected === null ||
      !Number.isInteger(this.selected) ||
      this.selected < 0 ||
      this.selected >= SeriesList.length
    ) {
      return 'シリーズから探す'
    }
    return getSeriesName(SeriesList[this.selected])
  }

  public get message() {
    return this.selected === null
      ? 'シリーズを選択してください'
      : this.songs.length === 0
      ? 'Not Found'
      : `Found ${this.songs.length} songs`
  }

  async asyncData({ params }: Context) {
    const index = Number.parseInt(params.id)
    if (isNaN(index) || index >= SeriesList.length) {
      return {
        isLoading: false
      }
    }
    const selectedSeries = SeriesList[index]
    try {
      const songs = await fetchSongs('series', selectedSeries)
      return {
        selected: index,
        songs,
        isLoading: false
      }
    } catch (e) {
      return {
        isLoading: false
      }
    }
  }
}
</script>
