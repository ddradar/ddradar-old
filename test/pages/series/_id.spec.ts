import {
  createLocalVue,
  mount,
  RouterLinkStub,
  shallowMount,
  Wrapper
} from '@vue/test-utils'
import Buefy from 'buefy'
import { mocked } from 'ts-jest/utils'

import SeriesPage from '@/pages/series/_id.vue'
import * as repo from '@/plugins/song-repository'
import { SeriesList } from '@/types/series'

jest.mock('~/plugins/firebase', () => {
  return {
    firestore: jest.fn()
  }
})
jest.mock('@/plugins/song-repository')

const localVue = createLocalVue()
localVue.use(Buefy)

describe('/series/:id', () => {
  let wrapper: Wrapper<SeriesPage>
  let vm: any
  beforeEach(() => {
    wrapper = shallowMount(SeriesPage, { localVue, stubs: ['song-list'] })
    vm = wrapper.vm
  })
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  describe('renders', () => {
    test('select SongIndex', () => {
      const wrapper = mount(SeriesPage, {
        localVue,
        stubs: { SongList: true, NuxtLink: RouterLinkStub }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('correctly', () => {
      const wrapper = mount(SeriesPage, {
        localVue,
        data: () => {
          return { selected: 10, songs: [{}] }
        },
        stubs: { SongList: true, NuxtLink: RouterLinkStub }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('not found', () => {
      const wrapper = mount(SeriesPage, {
        localVue,
        data: () => {
          return { selected: 10, songs: [] }
        },
        stubs: { SongList: true, NuxtLink: RouterLinkStub }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  test('selected:null and isLoading:true default', () => {
    expect(vm.selected).toBeNull()
    expect(vm.isLoading).toBe(true)
  })
  describe('asyncData method', () => {
    const hasDataIndex = 0
    const noDataIndex = 1
    const throwErrorIndex = 2
    beforeAll(() => {
      mocked(repo).fetchSongs.mockImplementation((_name, series) => {
        if (series === SeriesList[hasDataIndex]) {
          return Promise.resolve([
            {
              id: '00000000000000000000000000000000',
              name: 'PARANOiA',
              nameKana: 'PARANOIA',
              nameIndex: 25,
              artist: '180',
              series: '1st',
              minBPM: 180,
              maxBPM: 180,
              version: 20200101
            }
          ])
        }
        if (series === SeriesList[noDataIndex]) {
          return Promise.resolve([])
        }
        return Promise.reject(new Error('Mock test error'))
      })
    })
    test('selected is null if param is not 0-16', async () => {
      for (const param of ['', 'hoge', '37', '-1', 'NaN']) {
        const data = await vm.$options.asyncData({
          params: { id: param }
        })
        wrapper.setData(data)
        expect(vm.selected).toBeNull()
        expect(vm.isLoading).toBe(false)
      }
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { id: hasDataIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(hasDataIndex)
      expect(vm.isLoading).toBe(false)
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { id: noDataIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(noDataIndex)
      expect(vm.isLoading).toBe(false)
    })
    test('selected equals param if param is 0-16', async () => {
      const data = await vm.$options.asyncData({
        params: { id: throwErrorIndex.toString() }
      })
      wrapper.setData(data)
      expect(vm.selected).toBe(throwErrorIndex)
      expect(vm.isLoading).toBe(false)
    })
  })
  describe('pageTitle getter', () => {
    test('returns "シリーズから探す" if series not selected', () => {
      expect(vm.pageTitle).toBe('シリーズから探す')
    })
    test('returns "シリーズから探す" if selected is out of range', () => {
      for (const i of [-1, 100, 1.5, NaN, Infinity]) {
        wrapper.setData({ selected: i })
        expect(vm.pageTitle).toBe('シリーズから探す')
      }
    })
    test('returns Series title if selected', () => {
      wrapper.setData({ selected: 0 })
      expect(vm.pageTitle).toBe(`DDR ${SeriesList[0]}`)
      wrapper.setData({ selected: SeriesList.length - 1 })
      expect(vm.pageTitle).toBe(`DDR ${SeriesList[SeriesList.length - 1]}`)
    })
  })
  describe('message getter', () => {
    test('returns "シリーズを選択してください" if series not selected', () => {
      expect(vm.message).toBe('シリーズを選択してください')
    })
    test('returns "Not Found" if songs is empty', () => {
      wrapper.setData({ selected: 0, songs: [] })
      expect(vm.message).toBe('Not Found')
    })
    test('returns "Found n songs" songs is not empty', () => {
      wrapper.setData({
        selected: 0,
        songs: [
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          }
        ]
      })
      expect(vm.message).toBe('Found 1 songs')
      wrapper.setData({
        selected: 0,
        songs: [
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          },
          {
            name: 'PARANOiA',
            nameKana: 'PARANOIA',
            nameIndex: 25,
            artist: '180',
            series: '1st',
            minBPM: 180,
            maxBPM: 180
          }
        ]
      })
      expect(vm.message).toBe('Found 2 songs')
    })
  })
})
