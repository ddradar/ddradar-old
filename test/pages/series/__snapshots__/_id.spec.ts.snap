// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/series/:id renders correctly 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    DDR X
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      1st
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      3rdMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      4thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      5thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      EXTREME
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA2
    
      </span>
       
      <!---->
    </a>
    <button
      class="button is-info is-outlined"
      disabled="disabled"
      to="/series/10"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </button>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X3 VS 2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2013
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2014
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A20
    
      </span>
       
      <!---->
    </a>
  </div>
   
  <p
    class="subtitle"
  >
    Found 1 songs
  </p>
   
  <songlist-stub
    loading="true"
    songs="[object Object]"
  />
</section>
`;

exports[`/series/:id renders not found 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    DDR X
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      1st
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      3rdMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      4thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      5thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      EXTREME
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA2
    
      </span>
       
      <!---->
    </a>
    <button
      class="button is-info is-outlined"
      disabled="disabled"
      to="/series/10"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </button>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X3 VS 2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2013
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2014
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A20
    
      </span>
       
      <!---->
    </a>
  </div>
   
  <p
    class="subtitle"
  >
    Not Found
  </p>
   
  <songlist-stub
    loading="true"
    songs=""
  />
</section>
`;

exports[`/series/:id renders select SongIndex 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    シリーズから探す
  </h1>
   
  <div
    class="buttons"
  >
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      1st
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      3rdMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      4thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      5thMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      DDRMAX2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      EXTREME
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      SuperNOVA2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X2
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      X3 VS 2ndMIX
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2013
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      2014
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A
    
      </span>
       
      <!---->
    </a>
    <a
      class="button is-info"
      type="button"
    >
      <!---->
       
      <span>
        
      A20
    
      </span>
       
      <!---->
    </a>
  </div>
   
  <p
    class="subtitle"
  >
    シリーズを選択してください
  </p>
   
  <!---->
</section>
`;
